networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java

```diff
@@ -128,29 +131,47 @@ public Eth2P2PNetwork build() {
     // Build core network and inject eth2 handlers
     final DiscoveryNetwork<?> network = buildNetwork(gossipEncoding);
 
+    final GossipForkManager gossipForkManager = buildGossipForkManager(gossipEncoding, network);
+
     return new ActiveEth2P2PNetwork(
         config.getSpec(),
         asyncRunner,
-        metricsSystem,
         network,
         eth2PeerManager,
+        gossipForkManager,
         eventChannels,
         recentChainData,
         attestationSubnetService,
         gossipEncoding,
         config.getGossipConfigurator(),
-        gossipedBlockProcessor,
-        gossipedAttestationConsumer,
-        gossipedAggregateProcessor,
-        gossipedAttesterSlashingConsumer,
-        attesterSlashingGossipPublisher,
-        gossipedProposerSlashingConsumer,
-        proposerSlashingGossipPublisher,
-        gossipedVoluntaryExitConsumer,
-        voluntaryExitGossipPublisher,
         processedAttestationSubscriptionProvider);
   }
 
+  private GossipForkManager buildGossipForkManager(
+      final GossipEncoding gossipEncoding, final DiscoveryNetwork<?> network) {
+    final GossipForkManager.Builder gossipForkManagerBuilder =
+        GossipForkManager.builder().spec(spec).recentChainData(recentChainData);
+    gossipForkManagerBuilder.fork(
+        new GossipForkSubscriptionsPhase0(
+            spec.getForkManifest().get(UInt64.ZERO),
```

Comment:
Instead of hardcoding the phase0 subscriptions to the epoch0 fork, I think it would be more correct / less error-prone to provide the fork version:
```suggestion
            spec.getConfig().getGenesisForkVersion(),
```

We can then map from version -> `Fork` in the `gossipForkManagerBuilder` using `ForkManifest`.